src/org/labkey/targetedms/query/QCFolderDisplayColumnFactory.java

```diff
@@ -0,0 +1,65 @@
+package org.labkey.targetedms.query;
+
+import org.labkey.api.data.ColumnInfo;
+import org.labkey.api.data.Container;
+import org.labkey.api.data.DataColumn;
+import org.labkey.api.data.DisplayColumn;
+import org.labkey.api.data.DisplayColumnFactory;
+import org.labkey.api.data.RenderContext;
+import org.labkey.api.security.User;
+import org.labkey.api.security.permissions.ReadPermission;
+import org.labkey.api.targetedms.TargetedMSService;
+import org.labkey.api.util.PageFlowUtil;
+import org.labkey.targetedms.TargetedMSManager;
+import org.labkey.targetedms.TargetedMSModule;
+import org.labkey.targetedms.TargetedMSRun;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class QCFolderDisplayColumnFactory implements DisplayColumnFactory
+{
+    public QCFolderDisplayColumnFactory()
+    {
+    }
+
+    @Override
+    public DisplayColumn createRenderer(ColumnInfo colInfo)
+    {
+        return new DataColumn(colInfo)
+        {
+            @Override
+            public void renderGridCellContents(RenderContext ctx, Writer out) throws IOException
+            {
+                final User user = ctx.getViewContext().getUser();
+                String serialNumber = String.valueOf(ctx.get("SerialNumber"));
```

Comment:
We discussed yesterday that this value won't be available if someone customizes the view and removes the column.

See my suggestion above of aliasing the serial number as the value of this column. Then you can use getBoundColumn().getValue(ctx) to get the value in this code without needing to worry about the addQueryFieldKeys() stuff

Reply:
Sounds good